### PR TITLE
Update code example from supabase.mdx (Astro & Supabase Guide)

### LINKED ISSUE #8798: ⚠️ Issue Supabase & Astro guide, missing prerender set to false on hybrid
When following Supabase & Astro guide, the code examples for the api routes that handle the authentication throw the following error:
```
Request.formData: Could not parse content as FormData.
```

Since the project is in hybrid mode, the routes need to specify 
```
export const prerender = false;
```
```[tasklist]
### Tasks
- [x] Update Astro & Supabase guide
```

### PR DESCRIPTION
**Pull request by daniel-moya**
#### Description (required)
When following Supabase & Astro guide, the code examples for the api routes that handle the authentication throw the following error:
```
Request.formData: Could not parse content as FormData.
```

Since the project is in hybrid mode, the routes need to specify 
```
export const prerender = false;
```

@rythmlord on discord

#### Related issues & labels (optional)

- Closes #8798

### CODE REVIEW

File: src/content/docs/en/guides/backend/supabase.mdx [hunk 1]
```diff
@@ -122,7 +122,7 @@ To add authentication to your project, you will need to create a few server endp
 - `POST /api/auth/signin`: to sign in a user.
 - `GET /api/auth/signout`: to sign out a user.
 
-Create these endpoints in the `src/pages/api/auth` directory of your project. Your project should now include these new files:
+Create these endpoints in the `src/pages/api/auth` directory of your project. Make sure to specify `export const prerender = false` to render the endpoints on demand. Your project should now include these new files:
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/supabase.mdx:125]:
```suggestion
Create these endpoints in the `src/pages/api/auth` directory of your project. If you are using `hybrid` rendering mode, you must specify `export const prerender = false` at the top of each file to render these endpoints on demand. Your project should now include these new files:
```
This is only for `hybrid` mode, not `server` mode, so we we should make that clear!

File: src/content/docs/en/guides/backend/supabase.mdx [hunk 2]
```diff
@@ -143,6 +143,7 @@ Create these endpoints in the `src/pages/api/auth` directory of your project. Yo
 `register.ts` creates a new user in Supabase. It accepts a `POST` request with the an email and password. It then uses the Supabase SDK to create a new user.
 
 ```ts title="src/pages/api/auth/register.ts"
+export const prerender = false;
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/supabase.mdx:146]:
```suggestion
// With `output: 'hybrid'` configured:
// export const prerender = false;
```
Maybe something like this for these lines? This line is not necessarily correct for every project.

**Review by sarah11918 — COMMENTED** [review]
Hi @daniel-moya , thanks for pointing out how to follow this recipe for `hybrid` mode!

The change isn't quite as straightforward as it looks, however, because what you're suggesting works *only* for `hybrid` mode, and not for `server` mode. The instructions on line 18 state only that SSR must be enabled, not which specific mode to use. Therefore, the same instructions can't work for both.

This leaves us with some options! I've suggested below how to adjust your proposal to accommodate the two different possibilities.

The other option is to target line line 18 itself and make it something like:

- An Astro project with SSR enabled to allow for server-rendered endpoints, which are the default in `server` mode and require opting out of prerendering in `hybrid` mode.

Or, we could combine both! What do you think? What feels helpful to you?

File: src/content/docs/en/guides/backend/supabase.mdx [hunk 3]
```diff
@@ -171,6 +172,7 @@ export const POST: APIRoute = async ({ request, redirect }) => {
 `signin.ts` signs in a user. It accepts a `POST` request with the an email and password. It then uses the Supabase SDK to sign in the user.
 
 ```ts title="src/pages/api/auth/signin.ts"
+export const prerender = false;
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/supabase.mdx:175]:
```suggestion
// With `output: 'hybrid'` configured:
// export const prerender = false;
```

File: src/content/docs/en/guides/backend/supabase.mdx [hunk 4]
```diff
@@ -206,6 +208,7 @@ export const POST: APIRoute = async ({ request, cookies, redirect }) => {
 `signout.ts` signs out a user. It accepts a `GET` request and removes the user's access and refresh tokens.
 
 ```ts title="src/pages/api/auth/signout.ts"
+export const prerender = false;
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/supabase.mdx:211]:
```suggestion
// With `output: 'hybrid'` configured:
// export const prerender = false;
```

**Review by sarah11918 — COMMENTED** [review]
Thanks again for this helpful callout when using `hybrid` mode @daniel-moya ! I've gone ahead and updated the content appropriately to account for the fact that you might or might not be using hybrid mode, not assuming it. I'm sure this will help the people who are using `hybrid` mode, and we're happy to welcome you to Team Docs! :partying_face:

**Comment by daniel-moya** [comment]
> Or, we could combine both! What do you think? What feels helpful to you?

Hi Sarah, thanks for checking out my PR, Sorry i wasn't able to see your comment on time. I found your update quite sharp and helpful. 💯 

> and we're happy to welcome you to Team Docs! 🥳

Thanks it would be an amazing opportunity, Could you please point me in the right direction on how can I be part of the Team Docs, or be considered for it ?

**Comment by sarah11918** [comment]
Hi @daniel-moya -- no steps to take, you'll be a Team Docs membger (officially) as soon as this PR is merged! Your avatar will show up in the [list of docs contributors](https://docs.astro.build/en/contribute/#our-contributors) shortly!

It's just a friendly name we use, but you are also welcome to join the Astro Discord and self-select the role `@team-docs` to give yourself a different colour in the Discord. And occassionally we will notify "Team Docs" when there are issues or contributions that we encourage the community to participate in! https://astro.build/chat

**Pull request merged**